### DISCUSSION TITLE
Where does Huly Host?

### DISCUSSION THREAD

**Discussion by ZeroOneJP** [hcengineering/platform#6389] [Q&A] [answered]
Hello,

I´m thinking about using huly for my team, but want to know where Huly host their servers.
As I´m from Germany we got a strict Data Policy called DSGVO that we have to meet. So it would be perfect if the servers were in germany or europe.

Cheers
Felix

**Comment by aplatoff** [comment, marked as answer]
Hi Felix,

Nice question, let me explain current situation and ongoing changes related to the topic:
* Current huly.io hosted in the United States (East)
* We're in the process of upgrading our architecture, which affect hosting in following way:
* * Databases moving to Germany (Hetzner), and later will be distributed across the globe (however we'll be able to control concrete workspace locality so can pin your data to Germany
* * Document moving to Cloudflare R2, which means for ordinary workspaces we do not control where data physically resides and it's up to cloudflare (however we have different buckets and give CF hints for region e.g. Wester Europe), anyway this is not guaranteed and only hint, so we should not relay on this for regulatory purposes
* * Cloudflare have _Jurisdictional Restrictions_ R2 feature, which we also employ internally. They have `eu` region as a restriction so we can restrict any workspace to European Union: https://developers.cloudflare.com/r2/reference/data-location/

Above, by document I mean any document (file) attached to anything in Huly, any images, and all collaborative content, such as Task description, Wiki page, etc.

So answer is **yes, we can keep your data fully inside European Union** with our servers in Germany and Cloudflare R2 Jurisdictional Restrictions. But this is not done yet: we're in process of finalising set up and moving from US cluster to this new highly distributed solution. Hope we'll done this within a month or so. Technically Huly users will be able to say (optionally) where they want to host data, and also choose if this region is just an optimisation hint, or a strict regulatory requirement.

Thank you, and please ask anything if interested to know more. In the meantime you can start using Huly on US servers, all your data will be moved to EU when we done.

Kind regards,
Andrey

**Reply by lilienblum** [reply]
@aplatoff FYI Herzner Object Storage is in beta, you might want to try it before moving to CF R2
https://docs.hetzner.com/storage/object-storage/faq/beta

**Comment by nolanlws** [comment]
Any updates on that? Has the migration taken place already?

**Comment by aplatoff** [comment]
Hello,

Partially, we migrated some workspaces. Definitely finish all in Feb, so
you have all servers in Germany, and also possibility to limit your
workspace R2 storage to EU only using CF r@ Jurisdiction support.

**Reply by nolanlws** [reply]
That's great news, will there be some sort of setting to do that inside of the huly app?

> possibility to limit your workspace R2 storage to EU only using CF r@ Jurisdiction support

Also, do / will you guys offer DPAs?

**Comment by aplatoff** [comment]
Yes, during workspace creation time users will be able to choose data
location and jurisdiction. For EU jurisdiction we will provide DPA. For
existing workspaces we can help to transfer data and enforce jurisdiction
rules manually on request.

**Comment by elijh** [comment]
I do not yet see this option when I create a new workspace. I haven't signed up for a paid tier yet. Is this only available in a paid tier, or is the feature still pending?

**Reply by aplatoff** [reply]
Plans changed (or more correctly, postponed). At this moment all data moved to EU (Germany, Hetzner hosted cluster). Once we set up clusters in other countries, this feature will be available.

**Reply by badstue** [reply]
So all data has been moved to EU? Because I am interested in huly as well. And the support for 24 hour :-)

**Reply by mgela** [reply]
Hello any updates?
